testing-modules/testng_command_line/src/test/java/com/baeldung/testing_modules/testng_command_line/DateSerializerServiceUnitTest.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.testing_modules.testng_command_line;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(testName = "Date Serializer")
+public class DateSerializerServiceUnitTest {
+
+    private DateSerializerService toTest;
+
+    public DateSerializerServiceUnitTest() {
+        super();
+    }
+
+    @BeforeClass
+    public void beforeClass() {
+        toTest = new DateSerializerService();
+    }
+
+    @Test(groups = { "thread-safe", "system"}, expectedExceptions = {NullPointerException.class})
+    void givenNullDate_whenSerializeDate_thenThrowsException() {
+        //given
```

Comment:
Remove these unnecessary comments.

I've seen this pattern before, and in my personal opinion it's an anti-pattern to put boilerplate comments like this in tests. In Baeldung articles, we avoid adding extra comments or redundant code anyway.

At some point, someone on the internet demonstrated the `// given` `// when` `//then` comments, and it's spread like a meme, but if you think about it, it doesn't SAY anything.